File: Extension Objects/Runtime Ports/_Source Codes (XNA & Java)/XNA/CRunAdvComment.cs
// commentor object - XNA port (All Platforms)
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using RuntimeXNA.Extensions;
using RuntimeXNA.Services;
using RuntimeXNA.RunLoop;
using RuntimeXNA.Sprites;
using RuntimeXNA.Conditions;
using RuntimeXNA.Actions;
using RuntimeXNA.Expressions;
using RuntimeXNA.Objects;
using RuntimeXNA.Params;
using RuntimeXNA.Frame;
using RuntimeXNA.OI;
using RuntimeXNA.Movements;

namespace RuntimeXNA.Extensions
{
    class CRunAdvComment : CRunExtension
    {
        const int CND_COMMENT = 0;
        const int CND_OBJCOMMENT = 1;
        const int CND_TXTREGCOMMENT = 2;
        const int CND_TXTREGNOTE = 3;
        const int CND_TXTREGREMINDER = 4;
        const int CND_TXTREGIMPORTANT = 5;
        const int CND_TXTREGAAE = 6;
        const int CND_TXTCAPCOMMENT = 7;
        const int CND_TXTCAPNOTE = 8;
        const int CND_TXTCAPREMINDER = 9;
        const int CND_TXTCAPIMPORTANT = 10; // Huh, no CND_LAST

        const int ACT_COMMENT = 0;
        const int ACT_TXTREGCOMMENT = 1;
        const int ACT_TXTREGNOTE = 2;
        const int ACT_TXTREGREMINDER = 3;
        const int ACT_TXTREGIMPORTANT = 4;
        const int ACT_TXTREGAAA = 5;
        const int ACT_TXTREGAAB = 6;
        const int ACT_TXTCAPCOMMENT = 7;
        const int ACT_TXTCAPNOTE = 8;
        const int ACT_TXTCAPREMINDER = 9;
        const int ACT_TXTCAPIMPORTANT = 10;

        public override int getNumberOfConditions()
        {
            return 11;
        }

        public override bool createRunObject(CFile file, CCreateObjectInfo cob, int version)
        {
            return true;
        }

        public override bool condition(int num, CCndExtension cnd)
        {
            switch (num)
            {
                case CND_COMMENT:
					return true;
				case CND_OBJCOMMENT:
					return true;
				case CND_TXTREGCOMMENT:
					return true;
				case CND_TXTREGNOTE:
					return true;
				case CND_TXTREGREMINDER:
					return true;
				case CND_TXTREGIMPORTANT:
					return true;
				case CND_TXTREGAAE:
					return true;
				case CND_TXTCAPCOMMENT:
					return true;
				case CND_TXTCAPNOTE:
					return true;
				case CND_TXTCAPREMINDER:
					return true;
				case CND_TXTCAPIMPORTANT:
					return true;
            }
            return false;//won't happen
        }

        public override void action(int num, CActExtension act)
        {
            switch (num)
            {
                case ACT_COMMENT:
					break;
				case ACT_TXTREGCOMMENT:
					break;
				case ACT_TXTREGNOTE:
					break;
				case ACT_TXTREGREMINDER:
					break;
				case ACT_TXTREGIMPORTANT:
					break;
				case ACT_TXTREGAAA:
					break;
				case ACT_TXTREGAAB:
					break;
				case ACT_TXTCAPCOMMENT:
					break;
				case ACT_TXTCAPNOTE:
					break;
				case ACT_TXTCAPREMINDER:
					break;
				case ACT_TXTCAPIMPORTANT:
					break;
            }
        }

        public override CValue expression(int num)
        {
            switch (num)
            {
                // no expressions
            }
            return new CValue(0);// (Usually) won't be used
        }
    }
}

File: Extension Objects/Runtime Ports/_Source Codes (XNA & Java)/Java/CRunAdvComment.java
import java.awt.*;
import java.io.*;
import Extensions.*;
import Services.*;
import RunLoop.*;
import Conditions.*;
import Actions.*;
import Expressions.*;
import java.util.ArrayList;

public class CRunAdvComment extends CRunExtension
{
    // <editor-fold defaultstate="collapsed" desc=" A/C/E Constants ">
    public static final int CNDCOMMENT = 0;
    public static final int CNDOBJCOMMENT = 1;
    public static final int CNDTXTREGCOMMENT = 2;
    public static final int CNDTXTREGNOTE = 3;
    public static final int CNDTXTREGREMINDER = 4;
    public static final int CNDTXTREGIMPORTANT = 5;
    public static final int CNDTXTREGAAE = 6;
    public static final int CNDTXTCAPCOMMENT = 7;
    public static final int CNDTXTCAPNOTE = 8;
    public static final int CNDTXTCAPREMINDER = 9;
    public static final int CNDTXTCAPIMPORTANT = 10;
    public static final int CNDLAST = 11;

    public static final int ACTCOMMENT = 0;
    public static final int ACTTXTREGCOMMENT = 1;
    public static final int ACTTXTREGNOTE = 2;
    public static final int ACTTXTREGREMINDER = 3;
    public static final int ACTTXTREGIMPORTANT = 4;
    public static final int ACTTXTREGAAA = 5;
    public static final int ACTTXTREGAAB = 6;
    public static final int ACTTXTCAPCOMMENT = 7;
    public static final int ACTTXTCAPNOTE = 8;
    public static final int ACTTXTCAPREMINDER = 9;
    public static final int ACTTXTCAPIMPORTANT = 10;

    // </editor-fold>

    public CRunAdvComment()
    {

    }
    public @Override int getNumberOfConditions()
    {
	    return CNDLAST;
    }
    public @Override boolean createRunObject(CBinaryFile file, CCreateObjectInfo cob, int version)
    {
        return false;
    }
    public @Override void destroyRunObject(boolean bFast)
    {

    }


    public @Override int handleRunObject()
    {
        return 0;
    }


    public @Override void displayRunObject(Graphics2D g2)
    {
    }
    public @Override boolean saveRunObject(DataOutputStream stream)
    {
        return true;
    }
    public @Override boolean loadRunObject(DataInputStream stream)
    {
        return true;
    }

    // Conditions
    // -------------------------------------------------
    public @Override boolean condition(int num, CCndExtension cnd)
    {
        switch (num)
        {
            case CNDCOMMENT:
                return true;
            case CNDOBJCOMMENT:
                return true;
            case CNDTXTREGCOMMENT:
                return true;
            case CNDTXTREGNOTE:
                return true;
            case CNDTXTREGREMINDER:
                return true;
            case CNDTXTREGIMPORTANT:
                return true;
            case CNDTXTREGAAE:
                return true;
            case CNDTXTCAPCOMMENT:
                return true;
            case CNDTXTCAPNOTE:
                return true;
            case CNDTXTCAPREMINDER:
                return true;
            case CNDTXTCAPIMPORTANT:
                return true;
        }
        return false;
    }

    // Actions
    // -------------------------------------------------
    public @Override void action(int num, CActExtension act)
    {
        switch (num)
        {
            case ACTCOMMENT:
                break;
            case ACTTXTREGCOMMENT:
                break;
            case ACTTXTREGNOTE:
                break;
            case ACTTXTREGREMINDER:
                break;
            case ACTTXTREGIMPORTANT:
                break;
            case ACTTXTREGAAA:
                break;
            case ACTTXTREGAAB:
                break;
            case ACTTXTCAPCOMMENT:
                break;
            case ACTTXTCAPNOTE:
                break;
            case ACTTXTCAPREMINDER:
                break;
            case ACTTXTCAPIMPORTANT:
                break;
        }
    }

    // Expressions
    // -------------------------------------------------
    public @Override CValue expression(int num)
    {
        switch (num)
        {

        }
        return null;
    }
}